From c10f51816f70dd97e857a18c885c228f129ec7fd Mon Sep 17 00:00:00 2001
Message-Id: <c10f51816f70dd97e857a18c885c228f129ec7fd.1425109419.git.chang-joon.lee@intel.com>
In-Reply-To: <ce15578fa48a21eff276ee8163d395e39785921c.1425109418.git.chang-joon.lee@intel.com>
References: <ce15578fa48a21eff276ee8163d395e39785921c.1425109418.git.chang-joon.lee@intel.com>
From: Sonika Jindal <sonika.jindal@intel.com>
Date: Tue, 27 Jan 2015 17:58:21 +0530
Subject: [PATCH 3/4] FOR_UPSTREAM [VPG]: drm/i915/cht: Ignore short pulse

On HDMI hotplug interrupt, there are few spurious short pulses generated
by the interrupt controller. So schedule bottom half only when we receive
a long pulse, and ignore any short pulses.

Issue: GMINL-5482
Change-Id: Iec7a375a517b9c8f3351d980941a8f23fe77c40c
Signed-off-by: Sonika Jindal <sonika.jindal@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    3 +--
 drivers/gpu/drm/i915/i915_reg.h |    5 ++++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 038b88e2..edb3013 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -2223,8 +2223,7 @@ static void i9xx_hpd_irq_handler(struct drm_device *dev)
 
 		if (IS_VALLEYVIEW(dev)) {
 
-			/* Ignore short pulse interrupts */
-			if (!(hotplug_trigger & HPD_SHORT_PULSE)) {
+			if (hotplug_trigger & HPD_LONG_PULSE) {
 
 				/*
 				 * Few display's cant set the status for long
diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index cf03d6b..65701ea 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -2909,7 +2909,10 @@ enum punit_power_well {
 						 PORTC_HOTPLUG_INT_STATUS | \
 						 PORTD_HOTPLUG_INT_STATUS)
 
-#define HPD_SHORT_PULSE	(1<<17)
+#define HPD_B_LONG_PULSE	(1<<18)
+#define HPD_C_LONG_PULSE	(1<<20)
+#define HPD_D_LONG_PULSE	(1<<22)
+#define HPD_LONG_PULSE	(HPD_B_LONG_PULSE | HPD_C_LONG_PULSE | HPD_D_LONG_PULSE)
 
 /* SDVO and HDMI port control.
  * The same register may be used for SDVO or HDMI */
-- 
1.7.9.5

